File: playground/steps/openAi/callOpenAiApi.step.ts
import { ApiRouteConfig, StepHandler } from '@motiadev/core'
import { z } from 'zod'

const inputSchema = z.object({ message: z.string({ description: 'The message to send to OpenAI' }) })
const responseSchema = z.object({ message: z.string({ description: 'The message from OpenAI' }) })

export const config: ApiRouteConfig = {
  type: 'api',
  name: 'Call OpenAI',
  description: 'Call OpenAI',
  path: '/openai',
  method: 'POST',
  emits: ['call-openai'],
  flows: ['openai'],
  queryParams: [
    { name: 'model', description: 'The model to use' },
    { name: 'temperature', description: 'The temperature to use' },
  ],
  bodySchema: inputSchema,
  responseBody: responseSchema,
}

export const handler: StepHandler<typeof config> = async (req, { logger, emit }) => {
  logger.info('[Call OpenAI] Received callOpenAi event', req)

  await emit({
    topic: 'call-openai',
    data: { message: req.body.message },
  })

  return {
    status: 200,
    body: { message: 'OpenAI response sent' },
  }
}

File: playground/steps/openAi/callOpenAiApi.step.rb
def config
  {
    type: 'api',
    name: 'Call OpenAI',
    description: 'Call OpenAI API',
    path: '/openai-ruby',
    method: 'POST', 
    emits: ['call-openai'],
    flows: ['openai']
  }
end

def handler(req, ctx)
  ctx.logger.info('[Call OpenAI] Received callOpenAi event', req)

  message = req.body["message"]
  if message.nil?
    ctx.logger.warn('Message not found in request body')
    return
  end

  ctx.emit({
    "type" => "call-openai",
    "data" => { "message" => message }
  })

  {
    "status" => 200,
    "body" => { "message" => "OpenAI request sent, RUBY" }
  }
end
